man/sd_bus_add_object_vtable.xml

```diff
@@ -220,7 +240,45 @@
     for the callback functions (offset by the <parameter>offset</parameter> offsets as specified in
     the vtable entries).</para>
 
-    <para>For both functions, a match slot is created internally. If the output parameter
+    <para><function>sd_bus_add_object()</function> attaches a callback directly to the object path
+    <parameter>path</parameter>. An object path can have any number of callbacks attached to it.
+    Each callback is prepended to the list of callbacks which are always called in order.
+    <function>sd_bus_add_fallback()</function> is similar to
+    <function>sd_bus_add_object()</function> but applies to fallback paths instead.</para>
+
+    <para>When a request is received, any associated callbacks are called sequentially until a
+    callback returns a non-zero integer. Return zero from a callback to defer handling of the
+    request to the next callback. Callbacks are called in the following order: first, callbacks
+    attached directly to the request object path are called, followed by any D-Bus method callbacks
+    attached to the request object path, interface and member. Finally, the property callbacks
+    attached to the request object path, interface and member are called. If the final callback
+    returns zero, an error reply is sent back to the caller indicating no matching object for the
```

Comment:
Duh, this is super complicated. It would be great to have an example that shows all the possible ways to attach things and explains the priority. Doesn't have to be in this PR of course.

Reply:
I feel like we're getting dangerously close to having too much general information stuffed in the add_object_vtable manpage. If I had the time, I think something like "The systemd manual" wouldn't hurt to have (the [buildroot manual](https://buildroot.org/downloads/manual/manual.html) is a good example). The man pages are great as a reference but it'd be great to have a higher level manual that explains the higher level systemd design and concepts, introduces all the available tooling associated with each of these concepts and explains how to write daemons that take full advantage of everything systemd has to offer. I think that would also be a great place to put a higher level design overview of sd-bus including the example on callback ordering you mention. 

Of course, that's an extremely long-term goal at the rate I'm writing new and improving the existing man pages so let's see how far I get with those first.

Reply:
>  I think something like "The systemd manual" wouldn't hurt to have

Yeah, that'd be great. But that'd be a huge effort, and I think it'd quickly turn into "administration of linux systems with systemd". But on a smaller scale, a manual for sd-bus use would be quite useful and achievable I think. Most of the big functions are now documented, so what remains to be done is to describe some concepts that fell in between the cracks, like the callback ordering and logic, how filtering works, etc, and then provide a guide how to build a service using all that. FWIW, dnf plans to start using sd-bus: https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/message/3FENOBQVHKYPWENJ6YTPIEJOPPP5PBZZ/.